### PR TITLE
imbeats: harden compressed frame parsing

### PR DESCRIPTION
**Pull request by rgerhards**
## Summary

- reject malformed compressed Lumberjack payloads as protocol input errors
- reject trailing bytes after a zlib stream, nested compressed frames, zero-length nested JSON, and decompressed event overflow
- count compressed rejects through existing imbeats counters
- add regression tests for malformed compressed frames and compressed window overflow

## Validation

- `git diff --check`
- `bash -n tests/imbeats-compressed-window-overflow.sh tests/imbeats-compressed-malformed.sh`
- previously validated before this PR split with `autogen.sh`, `make -j$(nproc) check TESTS=""`, focused imbeats tests, and mock `distcheck`

### CODE REVIEW

File: tests/imbeats-compressed-window-overflow.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+. ${srcdir:=.}/diag.sh init
+require_plugin imbeats
+
+generate_conf
+add_conf '
+module(load="../plugins/impstats/.libs/impstats" interval="1"
```
[the remark this answers is not in the document]
Comment (rgerhards) [reply tests/imbeats-compressed-window-overflow.sh:8]: Already addressed in the current head: the test now has `require_plugin impstats` before loading impstats.
